src/components/Calendar.tsx

```diff
@@ -91,25 +121,63 @@ export const Calendar = ({ query, queryVariables, calendarData, isLoading, navig
     [navigation, query, queryVariables, calendarData, contentContainerId]
   );
 
+  const buildListItems = useCallback(
+    (data: DateData) => {
+      const listItems = parseListItemsFromQuery(query, data, { queryVariableWithDateRange });
+
+      return listItems;
+    },
+    [query, queryVariableWithDateRange]
+  );
+
   return (
-    <RNCalendar
-      dayComponent={DayComponent}
-      onDayPress={onDayPress}
-      displayLoadingIndicator={isLoading}
-      markedDates={getMarkedDates(calendarData)}
-      markingType="multi-dot"
-      renderArrow={renderArrow}
-      firstDay={1}
-      theme={{
-        todayTextColor: colors.primary,
-        indicatorColor: colors.primary,
-        dotStyle: {
-          borderRadius: DOT_SIZE / 2,
-          height: DOT_SIZE,
-          width: DOT_SIZE
-        }
-      }}
-      enableSwipeMonths
-    />
+    <>
+      <RNCalendar
+        dayComponent={DayComponent}
+        onDayPress={onDayPress}
+        displayLoadingIndicator={isLoading}
+        markedDates={markedDates}
+        markingType="multi-dot"
+        renderArrow={renderArrow}
+        firstDay={1}
+        theme={{
+          todayTextColor: colors.primary,
+          indicatorColor: colors.primary,
+          dotStyle: {
+            borderRadius: DOT_SIZE / 2,
+            height: DOT_SIZE,
+            width: DOT_SIZE
+          }
+        }}
+        enableSwipeMonths
+      />
+
+      {query === QUERY_TYPES.EVENT_RECORDS && subList && !loading ? (
```

Comment:
if we request `data` for `QUERY_TYPES.EVENT_RECORDS` in

```
const { data, loading } = useQuery(getQuery(QUERY_TYPES.EVENT_RECORDS), {
  variables: queryVariableWithDateRange
});
```

do we really need to check the type here again? doesn't `subList && !loading` check enough what we want to know for rendering list items?

Reply:
i think it would be more readable and usable to only check for `subList` here and let `ListEmptyComponent` handle the `loading` state. otherwise `loading` can never be true there, right?
we can also get rid of the else part which made think about too much.
```
 : (
    subList && (
      <LoadingContainer>
        <ActivityIndicator color={colors.accent} />
      </LoadingContainer>
    )
  )}
```

Reply:
Yes, you are right. We don't need the extra control.

The reason I added an extra else section is that sometimes the list does not empty. The list is always on the screen and the user may have a hard time realising whether the list has been refreshed or not. But this is not always the case. I am deleting the else section for now, but I think we will need to add it again in the future.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/34045f488b0635c5c06dffbb0f7dc8b139a98275